tests/deadcode.py

```diff
@@ -0,0 +1,269 @@
+
+        elif isinstance(node, ast.Match):
+            self.walk_deadcode(node.subject, deadcode)
+            for _case in node.cases:
+                _case.deadcode = deadcode
+                self.walk_deadcode(_case.pattern, deadcode)
+                self.walk_deadcode(_case.guard, deadcode)
+
+            dead_cases = all(
+                [self.check_stmts(_case.body, deadcode) for _case in node.cases]
+            )
+
+            if any(
+                isinstance(_case.pattern, ast.MatchAs) and _case.pattern.pattern is None
+                for _case in node.cases
+            ):
+                # case _:
+                deadcode = dead_cases
+
+        elif isinstance(node, (ast.For, ast.AsyncFor)):
+            self.walk_deadcode(node.target, deadcode)
+            self.walk_deadcode(node.iter, deadcode)
+            self.check_stmts(node.body, deadcode)
+            self.check_stmts(node.orelse, deadcode)
+
+        elif isinstance(node, ast.IfExp):
+
+            test_value = self.static_value(node.test, deadcode)
+
+            if_is_dead = self.walk_deadcode(
+                node.body, deadcode or (test_value == False)
+            )
+            else_is_dead = self.walk_deadcode(
+                node.orelse, deadcode or (test_value == True)
+            )
+
+        elif isinstance(node, (ast.While)):
+            cnd = self.static_value(node.test, deadcode)
+
+            self.check_stmts(node.body, deadcode or cnd == False)
+            self.check_stmts(node.orelse, deadcode or cnd == False or cnd == True)
+
+            def contains_break(node):
+                "search all child nodes except other loops for a break statement"
+                for child in ast.iter_child_nodes(node):
+                    if isinstance(child, (ast.For, ast.While, ast.AsyncFor)):
+                        continue
+                    if isinstance(child, ast.Break):
+                        return True
+                    if contains_break(child):
+                        return True
+
+                return False
+
+            if self.static_value(node.test, deadcode) == True and not contains_break(
+                node
+            ):
+                # while True: ... no break
+                deadcode = True
+
+        elif isinstance(node, (ast.Try, ast.TryStar)):
+            try_dead = self.check_stmts(node.body, deadcode)
+
+            for handler in node.handlers:
+                self.walk_deadcode(handler.type, deadcode)
+
+            handlers_dead = all(
+                [self.check_stmts(h.body, deadcode) for h in node.handlers]
+            )
+            else_dead = self.check_stmts(node.orelse, try_dead)
+            final_dead = self.check_stmts(node.finalbody, deadcode)
+
+            deadcode = (handlers_dead and else_dead) or final_dead
+
+        else:
+
+            self.static_value(node, deadcode)
+
+        return deadcode
+
+
+if __name__ == "__main__":
+    import sys
```

Comment:
Thanks for this main script, it's very useful to check what the code does. Please wrap this code into a `main()` function to avoid name shadowing.